File: Paradigmas de linguagem de programação/Lista de Exercicio - PLP/Questão 04/Go/main.go
package main
import "fmt"

type Animal interface {
    Som() string
}

type Cachorro struct{}

func (c Cachorro) Som() string {
    return "Cachorro: AuAu"
}

type Gato struct{}

func (g Gato) Som() string {
    return "Gato: Miau"
}

func FazerSom(a Animal) {
    fmt.Println(a.Som())
}

func main() {

    cachorro := Cachorro{}
    gato := Gato{}

    FazerSom(cachorro)
    FazerSom(gato)
}

File: Paradigmas de linguagem de programação/Lista de Exercicio - PLP/Questão 04/Java/Main.java
interface Animal {
    String som();
}

class Cachorro implements Animal {
    @Override
    public String som() {
        return "Cachorro: AuAu";
    }
}

class Gato implements Animal {
    @Override
    public String som() {
        return "Gato: Miau";
    }
}

public class Main {
    public static void fazerSom(Animal animal) {
        System.out.println(animal.som());
    }

    public static void main(String[] args) {
        Cachorro cachorro = new Cachorro();
        Gato gato = new Gato();

        fazerSom(cachorro);
        fazerSom(gato);
    }
}
